src/ert/_c_wrappers/job_queue/ext_job.py

```diff
@@ -1,263 +1,211 @@
-
-    def get_default_mapping(self) -> Dict[str, str]:
-        return dict(**self._get_default_mapping())
-
-    def set_environment(self, key: str, value: str):
-        self._set_environment(key, value)
-
-    def get_arglist(self) -> List[str]:
-        # We unescape backslash here to keep backwards compatability ie. If
-        # the arglist contains a '\n' we interpret it as a newline.
-        return [
-            s.encode("utf-8", "backslashreplace").decode("unicode_escape")
-            for s in self._get_arglist()
-        ]
-
-    def get_argvalues(self) -> List[str]:
-        return list(self._get_argvalues())
-
-    def set_arglist(self, args: List[str]):
-        return self._set_arglist(StringList(args))
-
-    def clear_environment(self):
-        self._clear_environment()
-
-    def save(self):
-        self._save()
-
-    def free(self):
-        self._free()
-
-    def name(self) -> str:
-        return self._get_name()
-
-    def __ne__(self, other) -> bool:
-        return not self == other
-
-    def __eq__(self, other) -> bool:
-        if not isinstance(other, ExtJob):
-            return False
-
-        if self.name() != other.name():
-            return False
-
-        if self.get_arglist() != other.get_arglist():
-            return False
+            path_to_executable = executable
+
+        resolved = None
+        if os.path.exists(path_to_executable):
+            resolved = path_to_executable
+        elif not os.path.isabs(executable):
+            # look for system installed executable
+            resolved = shutil.which(executable)
+
+        if resolved is None:
+            raise ConfigValidationError(
+                f"Could not find executable {executable} for job {name}"
+            )
 
-        if self.get_config_file() != other.get_config_file():
-            return False
+        if not os.access(resolved, os.X_OK):  # is not executable
+            raise ConfigValidationError(
+                f"ExtJob {name} with executable"
+                f" {resolved} does not have execute permissions"
+            )
 
-        if self.get_stderr_file() != other.get_stderr_file():
-            return False
+        if os.path.isdir(resolved):
+            raise ConfigValidationError(
+                f"ExtJob {name} has executable set to directory {resolved}"
+            )
 
-        if self.get_stdout_file() != other.get_stdout_file():
-            return False
+        return resolved
 
-        return True
+    def copy(self):
+        return replace(self)
+
+    _int_keywords = ["MAX_RUNNING", "MAX_RUNNING_MINUTES", "MIN_ARG", "MAX_ARG"]
+    _str_keywords = [
+        "STDIN",
+        "STDOUT",
+        "STDERR",
+        "START_FILE",
+        "EXECUTABLE",
+        "PORTABLE_EXE",
+        "TARGET_FILE",
+        "ERROR_FILE",
+        "START_FILE",
+    ]
+
+    @classmethod
+    def _parse_config_file(cls, config_file: str):
+        parser = ConfigParser()
```

Comment:
Is it possible to have this `parser = ConfigParser()` created only once or do we need to have a new one each time we call `ExtJob.from_config_file(...)`?

Reply:
one solution that i could envision is to create a parser that lives in the class - not sure if that is the best solution though, and also whether it goes against Eivind's dataclass / as little logic as possible approach...

Reply:
I guess this can be viewed as premature optimization. It is fine as it is